### PR TITLE
Remove uses of the Submariner clientset

### PR DESCRIPTION
**Pull request by skitt**
Controllers are supposed to use the client given to them by the
manager, instead of using generated clientsets. This removes our uses
of the Submariner clientset, except (for now) in tests.

Ultimately this will simplify the operator rebuild, since we won't
have to take care of rebuilding the clientset.

The clientset itself can't be removed yet, subctl imports it.

Signed-off-by: Stephen Kitt <skitt@redhat.com>

### CODE REVIEW

**Review by dfarrell07 — APPROVED** [review]

File: pkg/discovery/network/network.go [hunk 1]
```diff
@@ -63,7 +64,7 @@ func (cn *ClusterNetwork) IsComplete() bool {
 	return cn != nil && len(cn.ServiceCIDRs) > 0 && len(cn.PodCIDRs) > 0
 }
 
-func Discover(dynClient dynamic.Interface, clientSet kubernetes.Interface, operatorClient operatorclientset.Interface,
+func Discover(dynClient dynamic.Interface, clientSet kubernetes.Interface, operatorClient client.Client,
```
Comment (tpantelis) [inline pkg/discovery/network/network.go:67]: So this introduces yet another client API to callers in addition to the dynamic API and K8s generated API. Perhaps we should just simplify and use the `controller-runtime` API in lieu of the `kubernetes.Interface` as well. I think we still need the dynamic API when the Go type isn't available.

File: pkg/client/default_producer.go [hunk 2]
```diff
@@ -49,7 +47,3 @@ func (p *DefaultProducer) ForDynamic() dynamic.Interface {
 func (p *DefaultProducer) ForSubmariner() submarinerclient.Interface {
 	return p.SubmarinerClient
 }
-
-func (p *DefaultProducer) ForOperator() operatorclient.Interface {
```
Comment (tpantelis) [inline pkg/client/default_producer.go:53]: **subctl** uses this.

Comment (skitt) [reply pkg/client/default_producer.go:53]: Yup, I saw that too :-/.

File: pkg/discovery/network/network.go (hunk 1 above)
Comment (skitt) [reply pkg/discovery/network/network.go:67]: This is based on the `controller-runtime` pattern (where `Reconciler`s encapsulate a `client.Client` obtained from `mgr`).

Comment (tpantelis) [reply pkg/discovery/network/network.go:67]: Right - I suggest if we're going to switch to use the `controller-runtime` API in lieu of the operator client set then let's do the same for the other generated client sets. Also change the `client.Producer`.

File: pkg/client/default_producer.go (hunk 2 above)
Comment (mkolesnik) [reply pkg/client/default_producer.go:53]: Perhaps we need a test for subm-op to make sure were not breaking subctl

**Comment by Jaanki** [comment]
Please also remove the [make target](https://github.com/submariner-io/submariner-operator/blob/devel/Makefile#L183) to generate the clientset.

**Review by Jaanki — CHANGES_REQUESTED** [review]
Remove https://github.com/submariner-io/submariner-operator/blob/devel/Makefile#L183 target.

**Comment by skitt** [comment]
> Please also remove the [make target](https://github.com/submariner-io/submariner-operator/blob/devel/Makefile#L183) to generate the clientset.

I did this, but then reverted it because we still need to be able to create the clientset for `subctl`. I’ll drop the Makefile section when we can actually delete all the clientset code.

File: pkg/discovery/network/network.go (hunk 1 above)
Comment (skitt) [reply pkg/discovery/network/network.go:67]: I’ve made this PR specifically about the Submariner clientset, we can address the others in different PRs.

**Review by tpantelis — APPROVED** [review]

**Comment by tpantelis** [comment]
> > Please also remove the [make target](https://github.com/submariner-io/submariner-operator/blob/devel/Makefile#L183) to generate the clientset.
> 
> I did this, but then reverted it because we still need to be able to create the clientset for `subctl`. I’ll drop the Makefile section when we can actually delete all the clientset code.

We're going to need to adjust **subctl** anyway so I don't think we need to worry about restricting changes. But we can make the changes in multiple PRs then, once done, bump and adjust **subctl**.

**Review by Jaanki — APPROVED** [review]

**Pull request merged**